packages/block-library/src/query/edit/inspector-controls/format-controls.js

```diff
@@ -45,6 +45,20 @@ function formatNamesToValues( names, formats ) {
 }
 
 export default function FormatControls( { onChange, query: { format } } ) {
+	// 'format' is expected to be an array. If it is not an array, for example
+	// if a user has manually entered an invalid value in the block markup,
+	// convert it to an array to prevent JavaScript errors.
+	let formatArray = format;
+	if ( Array.isArray( format ) ) {
+		formatArray = format;
+	} else if ( typeof format === 'string' ) {
+		// If the format is a string, preserve it as an array.
+		formatArray = [ format ];
+	} else {
+		// If the format is not a string or an array, set it to an empty array.
+		formatArray = [];
+	}
```

Comment:
I think this can be oneliner; the editor often does similar normalizations.

```js
const normalizedFormats = Array.isArray( format ) ? format : [ format ];
```

Reply:
What if format is neither a string or array?

Reply:
Well either way it will validated against the available formats later.

Reply:
> Well either way it will validated against the available formats later.

Exactly. Also, if someone really wants to break to block, it's probably better to show them an error. Makes debugging easier